homeassistant/components/smappee/config_flow.py

```diff
@@ -21,3 +25,124 @@ class SmappeeFlowHandler(
     def logger(self) -> logging.Logger:
         """Return logger."""
         return logging.getLogger(__name__)
+
+    async def async_step_zeroconf(self, discovery_info):
+        """Handle zeroconf discovery."""
+
+        if not discovery_info[CONF_HOSTNAME].startswith("Smappee1"):
+            # We currently only support Energy and Solar models (legacy)
+            return self.async_abort(reason="invalid_mdns")
+
+        serial_number = (
+            discovery_info[CONF_HOSTNAME].replace(".local.", "").replace("Smappee", "")
+        )
+
+        # Check if already configured
+        await self.async_set_unique_id(f"Smappee{serial_number}")
```

Comment:
For the local and zeroconf entries we don't need to add `Smappee` to the unique id. The config entries core is aware of the integration domain when comparing unique ids. Just use the serial number.